src/components/PrivateLayout/useOneClickOutside.tsx

```diff
@@ -0,0 +1,33 @@
+import { RefObject, useEffect } from 'react'
+
+type AnyEvent = MouseEvent | TouchEvent
+
+function useOnClickOutside<T extends HTMLElement = HTMLElement>(
+  ref: RefObject<T>,
+  handler: (event: AnyEvent) => void,
+) {
+  useEffect(() => {
+    const listener = (event: AnyEvent) => {
+      const el = ref?.current
```

Comment:
We don't need this variable

Reply:
you're right but we don't need this variable
  if (!ref?.current|| ref?.current.contains(event.target as Node)) {